core/trino-main/src/main/java/io/trino/execution/scheduler/FaultTolerantPartitioningSchemeFactory.java

```diff
@@ -66,7 +67,7 @@ private FaultTolerantPartitioningScheme create(PartitioningHandle partitioningHa
         if (partitioningHandle.getCatalogHandle().isPresent() ||
                 (partitioningHandle.getConnectorHandle() instanceof MergePartitioningHandle)) {
             // TODO This caps the number of partitions to the number of available nodes. Perhaps a better approach is required for fault tolerant execution.
-            BucketNodeMap bucketNodeMap = nodePartitioningManager.getNodePartitioningMap(session, partitioningHandle).asBucketNodeMap();
+            BucketNodeMap bucketNodeMap = nodePartitioningManager.getNodePartitioningMap(session, partitioningHandle, getFaultTolerantExecutionPartitionCount(session)).asBucketNodeMap();
```

Comment:
Was this a bug since `io.trino.sql.planner.NodePartitioningManager#systemNodePartitionMap` did use `getHashPartitionCount(session)`? cc @arhimondr